### PR TITLE
Add pre-commit workflow

### LINKED ISSUE #313: Implement pre-commit hooks as status checks
This repository has [pre-commit](https://github.com/metal3-io/metal3-io.github.io/blob/source/GUIDELINES.md#technical-setup) checks that are supposed to be run before pushing and creating a PR. However, it requires the user to be aware about this and to actually enable/install it locally. Otherwise no check is done.
To avoid mistakes, we should add these pre-commit checks as status checks for PRs in this repository. Then it would be impossible to merge content that wasn't checked.

### PR DESCRIPTION
**Pull request by lentzi90**
Fixes #313

### CODE REVIEW

File: GUIDELINES.md [hunk 1]
```diff
@@ -123,7 +124,7 @@ All Blog posts are located in the [blog/\_posts](blog/_posts/) directory, and al
 
 ### Pages
 
-_[Pages](https://jekyllrb.com/docs/pages/)_ are located at the [pages](/pages/) directory, to create one follow these steps:
+The _[Pages](https://jekyllrb.com/docs/pages/)_ are located at the [pages](/pages/) directory, to create one follow these steps:
```
Comment (lentzi90) [inline GUIDELINES.md:127]: Added "The" to make it stop complaining about "repeated word". Same below for "Labs".

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  push:
+    branches: [source]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+    - uses: pre-commit/action@v3.0.0
```
Comment (tuminoid) [inline .github/workflows/pre-commit.yml:14]: This action is in maintenance mode. It is still maintained but they are advising using `pre-commit.ci`. Did you check that?

Comment (lentzi90) [reply .github/workflows/pre-commit.yml:14]: I checked briefly. It looks like a full blown CI system with fancy stuff that I don't think we want. The tests would run through their SaaS and fix any issues that can be fixed automatically.

> if tools make changes to files during a pull request, pre-commit.ci will automatically fix the pull request.

I would say we can continue with the pre-commit action. It is maintained, but any new feature requests are basically directed to pre-commit.ci.

Comment (tuminoid) [reply .github/workflows/pre-commit.yml:14]: Yeah it looked like that to me as well. It is still good to acknowledge that this action might not be maintained for too long.

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Pull request merged**